src/agent/src/device.rs

```diff
@@ -762,6 +836,45 @@ async fn vfio_device_handler(device: &Device, sandbox: &Arc<Mutex<Sandbox>>) ->
     })
 }
 
+// The VFIO AP (Adjunct Processor) device handler takes all the APQNs provided as device options
+// and awaits them. It sets the minimum AP rescan time of 5 seconds and temporarily adds that
+// amoutn to the hotplug timeout.
+#[cfg(target_arch = "s390x")]
+#[instrument]
+async fn vfio_ap_device_handler(
+    device: &Device,
+    sandbox: &Arc<Mutex<Sandbox>>,
+) -> Result<SpecUpdate> {
+    for ap_opt in device.options.iter() {
+        let opt = ap_opt.strip_prefix("ap:").ok_or_else(|| {
+            anyhow!(
+                "VFIO AP device option {} unexpectedly missed `ap:` indication",
+                ap_opt
+            )
+        })?;
+
+        let mut config = AGENT_CONFIG.write().await;
+        let hotplug_timeout = config.hotplug_timeout;
+        warn!(sl!(), "Temporarily adding {} seconds to hotplug timeout for AP devices due to rescan intervals", AP_CONFIG_TIME_SECONDS);
+        config.hotplug_timeout += time::Duration::from_secs(AP_CONFIG_TIME_SECONDS as u64 + 5);
```

Comment:
You're doing `+=` so why the `+ 5` at the end? I'm guessing that was the original value before you added `AP_CONFIG_TIME_SECONDS`?

Reply:
Oops, that is not necessary and was added in development in confusion over what turned out to be another bug. Will remove.

Reply:
Yikes.  Making the global config writable just for this seems like a terrible idea to me.  That seems really likely to break people's assumptions that it's essentially a write-once global.